### PR TITLE
v2 transfer

### PR DESCRIPTION
**Pull request by hourglasshoro**
# Overview
* Implement transfer
* Add test
* contract size check in CI

### CODE REVIEW

File: test/mint.js [hunk 1]
```diff
@@ -447,4 +447,55 @@ describe('PodCastNFT', function () {
...
+      await transferTx.wait()
+
+      expect(await contract.balanceOf(alice.address)).to.eq(0)
+      expect(contract.roles(alice.address, 0)).to.be.reverted
+      expect(contract.userAttribute(alice.address)).to.be.reverted
+      expect(await contract.balanceOf(bob.address)).to.eq(1)
+      expect(await contract.roles(bob.address, 0)).to.eq(roles[0])
+      expect(await contract.roles(bob.address, 1)).to.eq(roles[1])
+      expect((await contract.userAttribute(bob.address)).point).to.eq(point)
+      expect((await contract.userAttribute(bob.address)).claimableToken).to.eq(
```
Comment (hourglasshoro) [inline test/mint.js:496]: [IMO] I noticed while implementing the test that there is no need to increase the claimableToken in the giveAwayPoint?

Comment (SamuraiT) [reply test/mint.js:496]: @geeknees can you fix this ?
https://github.com/henkaku-center/podcast-nft/pull/52/files/bd8d9c8e94ee6dd1e675f2137972777da7238233#diff-091a64f3ece8473484ef0f9f24718ed5e3f9e669991fa082bbbc30719df20459

File: contracts/PodCastNFT.sol [hunk 2]
```diff
@@ -240,6 +251,14 @@ contract PodCastNFT is ERC721, Ownable {
         emit ClaimedToken(msg.sender, amount);
     }
 
+    function transfer(
+        address _to,
+        uint256 _tokenId,
+        bytes memory _data
+    ) public onlyHolder(msg.sender) hasTokenId(_tokenId) {
+        _safeTransfer(msg.sender, _to, _tokenId, _data);
+    }
```
Comment (SamuraiT) [inline contracts/PodCastNFT.sol:260]: you dont need this transfer function. 
erc721 has it's own transferFrom which uses `_afterTokenTransfer` inside of the function

File: .github/workflows/pull_request.yaml [hunk 3]
```diff
@@ -25,7 +25,26 @@ jobs:
...
       - name: Test
         run: yarn test
+      - name: Check Target Contract Size
+        run: echo "\`\`\`\n $(yarn hardhat size-contracts)" > ./comments
+      - name: Post comments
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          URL: ${{ github.event.pull_request.html_url }}
+        run:
+          gh pr comment -F ./comments "${URL}"
```
Comment (SamuraiT) [inline .github/workflows/pull_request.yaml:50]: super !

File: contracts/PodCastNFT.sol (hunk 2 above)
Comment (hourglasshoro) [reply contracts/PodCastNFT.sol:260]: I certainly didn't need this!

**Comment by SamuraiT** [comment]
can you make transferFrom only available for owner ?
we discussed about abuse of nft and decided to restrict use of transfer

**Comment by hourglasshoro** [comment]
> can you make transferFrom only available for owner ?

OK! override and implement!

**Review by SamuraiT — APPROVED** [review]

**Pull request merged**